### ISSUE TITLE
SAS Tokens seem not to be authorized.

### ISSUE THREAD

**Issue by thisisthekap** [arafato/azurite#124]
I am creating a SAS Token in this way:

```csharp
SharedAccessBlobPolicy sasConstraints = new SharedAccessBlobPolicy
{
	SharedAccessExpiryTime = DateTimeOffset.UtcNow.AddMinutes(30),
	Permissions = SharedAccessBlobPermissions.Read |
						SharedAccessBlobPermissions.Add |
						SharedAccessBlobPermissions.Write |
						SharedAccessBlobPermissions.Create |
						SharedAccessBlobPermissions.Delete
};
string sasToken = container.GetSharedAccessSignature(sasConstraints);
```

When I am using the token in a url to get a blob, I get `403 not found`.

The URL I am using looks like this:

`http://localhost:10000/devstoreaccount1/my-container/my-xml-file.xml?sv=2017-04-17&sr=c&sig=gAoe3OE9bgDfsFEFv0x5LHk0ifRYT4Yz0A1Jlbx9x6c%3D&se=2017-12-02T10%3A49%3A12Z&sp=racwd`

When using the Azure production environment everything works just fine.

**Comment by arafato** [comment]
Hi @thisisthekap , thanks for the report!
I've tried to repro but haven't been successful. In particular, I've tried to generate the signature based on your request (both with .NET SDK 8.6.0, and Node 2.6.0). For .NET, I used the following program with your provided values:
```
class Program
    {
        static void Main(string[] args)
        {
            CloudStorageAccount storageAccount = CloudStorageAccount.DevelopmentStorageAccount;
            CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
            CloudBlobContainer container = blobClient.GetContainerReference("my-container");
            var blob = container.GetBlobReference("my-xml-file.xml");

            SharedAccessBlobPolicy sasConstraints = new SharedAccessBlobPolicy
            {
                SharedAccessExpiryTime = DateTimeOffset.Parse("2017-12-02T10:49:12Z"),
                Permissions = SharedAccessBlobPermissions.Read |
                        SharedAccessBlobPermissions.Add |
                        SharedAccessBlobPermissions.Write |
                        SharedAccessBlobPermissions.Create |
                        SharedAccessBlobPermissions.Delete
            };
            Console.WriteLine(container.GetSharedAccessSignature(sasConstraints));
            Console.Read();
        }
    }
```
The signature being generated is (both .NET and Azurite)
`?sv=2017-04-17&sr=c&sig=zK0sDJeUfFR%2FWysug0el17zpv%2FFQ%2F%2BGsUErB2P0JTAs%3D&se=2017-12-02T10%3A49%3A12Z&sp=racwd`

As you can see, the query parameter values are equal to what you have posted, but the signature (sig param) is different from what you have posted. What kind of values / signature to you get if you execute above code? Are you happen to use your production key for signing the request?

**Comment by thisisthekap** [comment]
I am using the name and key provided by https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string
```
Account name: devstoreaccount1
Account key: Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==
```
And of course I get `403 forbidden` instead of `403 not found` (sorry for the mistake).

As mentioned, everything works just fine when I use my azure blob storage account incl. the key provided by Azure Portal.

**Comment by arafato** [comment]
@thisisthekap just found the bug. The code assumed that the start date must be explicitly set where it is optional. Just published v1.7.1 which should fix it.
